### PR TITLE
✨ Add literal operators to rest of VDML Devices

### PR DESCRIPTION
**Pull request by Gracelu128**
#### Summary:
Add literal operators for imu, gps, rotation, optical, serial, and distance.

##### References (optional):

#461 

#### Test Plan:
Try instantiating these devices with literals.
- [x] Imu
- [x] GPS
- [x] Rotation
- [x] Optical
- [x] Serial 
- [x] Distance

### CODE REVIEW

File: include/pros/optical.hpp [hunk 1]
```diff
@@ -251,6 +251,10 @@ class Optical {
 	const std::uint8_t _port;
 	///@}
 };
+
+namespace literals {
+const pros::Optical operator"" _o(const unsigned long long int o);
```
Comment (WillXuCodes) [inline include/pros/optical.hpp:256]: can we call this one _opt or something

File: src/devices/vdml_optical.cpp [hunk 2]
```diff
@@ -83,5 +83,10 @@ std::ostream& operator<<(std::ostream& os, pros::Optical& optical) {
   return os;
 }
 
+namespace literals {
+const pros::Optical operator"" _o(const unsigned long long int o) {
```
Comment (WillXuCodes) [inline src/devices/vdml_optical.cpp:87]: ditto here

File: src/devices/vdml_optical.cpp [hunk 3]
```diff
@@ -83,5 +83,10 @@ std::ostream& operator<<(std::ostream& os, pros::Optical& optical) {
...
+  return pros::Optical(o);
+}
+}
```
Comment (WillXuCodes) [inline src/devices/vdml_optical.cpp:90]: mind adding a newline at the end of the file here.

**Review by WillXuCodes — CHANGES_REQUESTED** [review]

File: src/devices/vdml_distance.cpp [hunk 4]
```diff
@@ -48,5 +48,10 @@ std::ostream& operator<<(std::ostream& os, pros::Distance& distance) {
     return os;
 }
 
+namespace literals {
+const pros::Distance operator"" _dist(const unsigned long long int d) {
+    return pros::Distance(d);
+}
+} // namespace literals
 }
 }  // namespace pros
```
Comment (WillXuCodes) [inline src/devices/vdml_distance.cpp:57]: mind adding a newline at the end of this file? Not your thing but it doesn't seem to exist on this file for some reason.

**Comment by WillXuCodes** [comment]
bump on hardware testing

**Review by Richard-Stump — APPROVED** [review]

File: src/devices/vdml_distance.cpp (hunk 4 above)
Comment (Richard-Stump) [reply src/devices/vdml_distance.cpp]: I added it

**Review by Richard-Stump — APPROVED** [review]

**Pull request merged**